src/attesters/api/enclave_attester_register.c

```diff
@@ -38,6 +38,16 @@ enclave_attester_err_t enclave_attester_register(const enclave_attester_opts_t *
 		}
 	}
 
+	if (opts->flags & ENCLAVE_ATTESTER_OPTS_FLAGS_SEV_GUEST) {
+		if (!is_sevguest_supported()) {
+			// clang-format off
+			RTLS_DEBUG("failed to register the attester '%s' due to lack of SEV(-ES) Guest capability\n",
+				   opts->type);
+			// clang-format on
+			return -ENCLAVE_ATTESTER_ERR_CPU_UNSUPPORTED;
+		}
+	}
+
 	enclave_attester_opts_t *new_opts = (enclave_attester_opts_t *)malloc(sizeof(*new_opts));
```

Comment:
Don't suggest to put self-detect function here. We can put it in its' init.c or pre_init.c.

Reply:
Good point! I suggest detecting the registered environment according to the previous logic and then modifying both sev, snp and tdx uniformly after the out-of-tree code is completed.